src/oracle/middleware.js

```diff
@@ -0,0 +1,761 @@
+
+        //if a request has not been made yet
+        if (allowedSend) {
+          //submit job
+          console.log("Initialising new CL job request")
+          submitNewJob(jobName, sendRequest)
+        }
+        else{
+          console.log("Will not be initialising new job request - Still waiting for request", query.request_id, "to finish. Last finished request is", query.last_received_request_id)
+        }
+      }
+    }
+  }, priceQueryInterval * 1_000);
+
+}
+
+/**
+  * Function to create a bridge which listens from the Chainlink node for
+  * new jobs, job removals and job run results
+  * @param {*} PORT the port to listen on
+  */
+const startBridge = (PORT) => {
+
+  console.log("Bridge started")
+  const app = express();
+  app.use(bodyParser.json());
+
+  /**
+  * POST /adapter endpoint
+  * This is used to listen for job run results
+  */
+  app.post('/adapter', async (req, res) => {
+    //get result
+    let result = Math.round(req.body.data.result)
+
+    //get run id and type
+    let requestId = String(req.body.data.request_id)
+    let requestType = Number(req.body.data.request_type)
+    let jobName = req.body.data.name
+    console.log("Bridge received " + String(result) + " for " + jobName + " (Request: " + requestId + ", Type: " + requestType + ")")
+
+    //return a 200 code to the Chainlink node if a successful price is found
+    if (!isNaN(result)){
+      res.status(200).send({ success: true })
+    } else {
+      res.status(500).send({ success: false })
+    } 
+
+    //get last price from state
+    let query = await queryTable("jobs", ["last_result"], jobName)
+    let lastPrice = query.last_result
+
+    //get push interval for feed
+    let pushInterval = feeds[jobName].pushInterval
+
+    //check if time for update
+    query = await queryTable("rounds", ["started_at"], jobName)
+
+    let timeForUpdate = (Date.now()/1000) >= query.started_at + Number(pushInterval)
+
+    //if there is no last price, if it is time for a price update or if there is a new round, update price
+    let toUpdate = lastPrice == -1 || lastPrice == 0 || requestType == 1 && timeForUpdate || requestType == 3
+    //if last price is found and it is a price deviation request
+    if (lastPrice != -1 && requestType == 2) {
+      
+      //get decimal places for feed
+      let decimalPlaces = feeds[jobName].decimalPlaces
+      //calculate percentage change
+      lastPrice = lastPrice * Math.pow(10, Number(decimalPlaces))
+
+      let percChange = Math.abs((result - lastPrice) / lastPrice) * 100
+      console.log("Price change is " + percChange + "%. Last Price: " + String(result) + ". Current Price: " + String(lastPrice))
+
+      //get price deviation threshold for feed
+      let priceDeviationPercentage = feeds[jobName].priceDeviationPerc
+      //update price if result is greater than price deviation threshold
+      toUpdate = percChange > priceDeviationPercentage
```

Comment:
You should this isn't >= ?

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/e9a33dabc173c4e8027eff5679d8e891ac7f5111